Neos.Flow/Classes/Persistence/Aspect/PersistenceMagicAspect.php

```diff
@@ -70,8 +70,8 @@ public function isEntityOrValueObject()
 
     /**
      * @var string
-     * @ORM\Id
-     * @ORM\Column(length=40)
+     * @Doctrine\ORM\Mapping\Id
```

Comment:
Curious, but doesn't the "manual" import in lines 14+15 suffice to have the short named annotation?

Reply:
No, because that docblock (of the aspect class) is not copied into the class the *property* is introduced into…

Reply:
Ah, right. Could we possibly monkey-patch https://github.com/neos/flow-development-collection/blob/master/Neos.Flow/Classes/Aop/PropertyIntroduction.php#L83 for a b/c fix to this? I guess it's still hard to resolve the string class name without parsing the `use` statements of the Aspect class... maybe at least resolve `@ORM\` and `@Flow\` there to somewhat lower breakyness?

Reply:
It did, because we did not copy the docblock, but re-rendered any annotations, making them fully qualified. As can be seen in the example with the `Entity` annotation in #2564.

Reply:
About the breakiness… TBH I have no clue how much property introduction is actually used and if those few places cannot be fixed faster, than it would take us to come up with a fix that doesn't come with additional (potential) bugs…

Reply:
https://github.com/search?l=PHP&p=8&q=%27%40Flow%5CIntroduce%27&ref=advsearch&type=Code is almost exclusively our (test) code.

Reply:
Okay - still, wdyt about making Flow and ORM work by hardcoding the FQCN resolution for those? Whenever you want to inject a property, it's most likely a DB field or some Flow injected property.

Reply:
That would reduce the pain. But for the record, in random projects I randomly don't use the general `Flow\Annotations` import, but rather import `@Inject()` (for example) directly, because Phpstorm let's me do that very easily. These cases would still break.

Reply:
> That would reduce the pain. But for the record, in random projects I randomly don't use the general `Flow\Annotations` import, but rather import `@Inject()` (for example) directly, because Phpstorm let's me do that very easily. These cases would still break.

uh, would _not_ break, right? Because I imported it and I'm not using the FQCN.

Reply:
**Would break**, because your import in the aspect class' docblock is never copied to the class you introduce a property to. This is _only about annotations on introduced properties_…

We can of course expand `@ORM\` and `@Flow`, feel free to improve this (I won't have time today)…